### ISSUE TITLE
Stopping on errors in subcommands (set -e;<cmd>)

### ISSUE THREAD

**Issue by springmeyer** [mapbox/mason#234]
I think @kkaefer fixed this in https://github.com/mapbox/mason/commit/fbc0781402c9f885dbf9170c54a7b4bd0ee541a3.

But I'm still seeing cases where things that should cause abrupt failures don't and mason continues along and reports and error code of 0. In the case below (where pkg-config fails) this silently creates a broken package. Without any non-zero error code the `mason.ini` myteriously lacks `include_dirs`.

```
$ ./mason build vectortile 1.0.0-rc4
* Removing existing package... /Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4
* Building for Platform 'osx/x86_64'...
* Installed at /Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4
Package /Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4/ was not found in the pkg-config search path.
Perhaps you should add the directory containing `/Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4/.pc'
to the PKG_CONFIG_PATH environment variable
No package '/Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4/' found
Package /Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4/ was not found in the pkg-config search path.
Perhaps you should add the directory containing `/Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4/.pc'
to the PKG_CONFIG_PATH environment variable
No package '/Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4/' found
* Linking vectortile 1.0.0-rc4 dynamically
* Wrote configuration file /Users/dane/.mason/mason_packages/headers/vectortile/1.0.0-rc4/mason.ini:
name=vectortile
version=1.0.0-rc4
header_only=true
$ echo $?
0

```

**Comment by springmeyer** [comment]
Need to fix this - otherwise it may result in quietly broken mason.ini (refs #370)

**Comment by springmeyer** [comment]
found it: the call the `mason_config` needs a `set -e` here: https://github.com/mapbox/mason/blob/bcbb1d98ff3f30cafd5b5591ecce882426b49d9c/mason.sh#L508. Working on a PR to fix.

**Comment by springmeyer** [comment]
Fixed in #373.
